File: src/dnd/items/Dagger.java
package dnd.items;

import dnd.combat.DamageDiceGroup;
import dnd.data.DamageType;
import dnd.data.WeaponProperty;

/**
 * A class representing the Dagger item
 * 
 * @author calvi
 *
 */
public class Dagger extends Item {

	/**
	 * Constructor for class Dagger
	 */
	public Dagger() {
		super("Dagger", new double[] { 20.0, 60.0 }, 5.0);
		weaponProperties.add(WeaponProperty.FINESSE);
		weaponProperties.add(WeaponProperty.LIGHT);
		weaponProperties.add(WeaponProperty.THROWN);
	}

	@Override
	public DamageDiceGroup getDamageDice() {
		return new DamageDiceGroup(1, 4, DamageType.PIERCING);
	}

}

File: scripts/items/dagger.lua
function define()
	DamageType = luajava.bindClass("com.dndsuite.dnd.data.DamageType")
	Item = luajava.bindClass("com.dndsuite.core.Item")
	ItemAttack = luajava.bindClass("com.dndsuite.core.events.ItemAttack")
	
	self:setName("Dagger")
	self:addTag(Item.FINESSE)
	self:addTag(Item.LIGHT)
	self:addTag(Item.SIMPLE_MELEE)
	self:addTag(Item.THROWN)
	self:addTag(Item.WEAPON)
	self:addTag(Item.DAGGER)
end



function acAbilityBonusLimit()
	return 0
end



function acbase()
	return 0
end



function customTasks()
end



function damage()
	if attackType == ItemAttack.MELEE then
		return luajava.newInstance("com.dndsuite.dnd.combat.DamageDiceGroup", 1, 4, DamageType.PIERCING)
	elseif attackType == ItemAttack.RANGED then
		return nil
	elseif attackType == ItemAttack.THROWN then
		return luajava.newInstance("com.dndsuite.dnd.combat.DamageDiceGroup", 1, 4, DamageType.PIERCING)
	end
end



function equip()
end



function range()
	return 20.0, 40.0, -1.0, -1.0
end



function reach()
	return 0.0
end



function unequip()
end



function weight()
	return 1.0
end